lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -602,26 +602,51 @@ def write(chunk)
             case error.code
-            when GRPC::Core::StatusCodes::CANCELLED,
-                 GRPC::Core::StatusCodes::UNAVAILABLE,
-                 GRPC::Core::StatusCodes::DEADLINE_EXCEEDED,
-                 GRPC::Core::StatusCodes::INTERNAL,
-                 GRPC::Core::StatusCodes::UNKNOWN
-              # Server error, so retry via re-raising the error.
+            # Server error, so retry via re-raising the error.
+            when \
+                # Maps to http status 500 (Internal Server Error)
+                GRPC::Core::StatusCodes::INTERNAL,
+                # Maps to http status 501 (Not Implemented).
+                GRPC::Core::StatusCodes::UNIMPLEMENTED,
+                # Maps to http status 503 (Service Unavailable).
+                GRPC::Core::StatusCodes::UNAVAILABLE,
+                # Maps to http status 504 (Gateway Timeout).
+                GRPC::Core::StatusCodes::DEADLINE_EXCEEDED
               increment_retried_entries_count(entries_count, error.code)
               @log.debug "Retrying #{entries_count} log message(s) later.",
                          error: error.to_s, error_code: error.code.to_s
               raise error
+
+            # Most client errors indicate a problem with the request itself and
+            # should not be retried.
             when \
-                # Most client errors indicate a problem with the request itself
-                # and should not be retried.
-                GRPC::Core::StatusCodes::UNIMPLEMENTED,
+                # Maps to http status 400 (Bad Request).
+                GRPC::Core::StatusCodes::INVALID_ARGUMENT,
+                # Authorization error. These are usually solved via a `gcloud
+                # auth` call, or by modifying the permissions on the Google
+                # Cloud project.
+                # Maps to http status 401 (Unauthorized).
+                GRPC::Core::StatusCodes::UNAUTHENTICATED,
+                # Maps to http status 403 (Forbidden).
+                GRPC::Core::StatusCodes::PERMISSION_DENIED,
+                # Maps to http status 404 (Not Found).
+                GRPC::Core::StatusCodes::NOT_FOUND,
+                # Maps to http status 409 (Conflict).
+                GRPC::Core::StatusCodes::ABORTED,
+                # Maps to http status 412 (Precondition Failed).
+                GRPC::Core::StatusCodes::FAILED_PRECONDITION,
+                # Maps to http status 429 (Too Many Requests).
                 GRPC::Core::StatusCodes::RESOURCE_EXHAUSTED,
-                # Authorization error.
-                # These are usually solved via a `gcloud auth` call, or by
-                # modifying the permissions on the Google Cloud project.
-                GRPC::Core::StatusCodes::UNAUTHENTICATED
+                # Maps to http status 499 (Client Closed Request).
+                GRPC::Core::StatusCodes::CANCELLED,
+                # Maps to the remaining http codes in both 4XX and 5XX category.
```

Comment:
Nit: `4xx and 5xx` (lowercase "x"). Also below.

Reply:
Changed.